php-frontend/src/main/java/org/sonar/php/utils/LiteralUtils.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.utils;
+
+public class LiteralUtils {
+
+  private LiteralUtils() {
+    // This class only contains static methods
+  }
+
+  public static Integer intLiteralValue(String literalValue) {
+    String value = literalValue.replace("_", "");
+    if (value.startsWith("0b") || value.startsWith("0B")) {
+      return Integer.parseUnsignedInt(value.substring(2), 2);
+    }
+    return Long.decode(value).intValue();

```

Comment:
Looking at https://www.php.net/manual/en/language.types.integer.php , it seems that literals can be 64 bit. I think it would be better to use `long` type to make sure we are able to parse bigger values like  `9223372036854775807`